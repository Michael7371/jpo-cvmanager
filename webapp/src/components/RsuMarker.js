import React from 'react'

function RsuMarker(props) {
  var circleStyle = {
    padding: 5,
    display: 'inline-block',
    borderRadius: '50%',
    width: 5,
    height: 5,
  }

  if (props.displayType === 'online') {
    if (props.onlineStatus === 'online') circleStyle.backgroundColor = '#A1D363'
    else if (props.onlineStatus === 'unstable') circleStyle.backgroundColor = '#D1A711'
    else if (props.onlineStatus === 'offline') circleStyle.backgroundColor = '#E94F37'
    else circleStyle.backgroundColor = '#B0B0B0'
  } else if (props.displayType === 'scms') {
    if (props.scmsStatus === '1') circleStyle.backgroundColor = '#A1D363'
    else if (props.scmsStatus === '0') circleStyle.backgroundColor = '#E94F37'
    else circleStyle.backgroundColor = '#B0B0B0'
  } else {
    circleStyle.backgroundColor = '#B0B0B0'
  }

  return <div style={circleStyle}></div>
}

export default RsuMarker
