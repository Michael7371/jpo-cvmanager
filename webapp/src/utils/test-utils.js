/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

export { replaceChaoticIds }
