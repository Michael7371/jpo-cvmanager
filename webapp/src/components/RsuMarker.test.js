import React from 'react'
import { render } from '@testing-library/react'
import RsuMarker from './RsuMarker'
import { replaceChaoticIds } from '../utils/test-utils'

it('should take a snapshot', () => {
  const { container } = render(<RsuMarker />)

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
