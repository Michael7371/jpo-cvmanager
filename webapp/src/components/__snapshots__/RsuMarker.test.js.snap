// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    style="padding: 5px; display: inline-block; border-radius: 50%; width: 5px; height: 5px; background-color: rgb(176, 176, 176);"
  />
</div>
`;
